opendevin/core/main.py

```diff
@@ -84,7 +91,10 @@ async def main(task_str: str = '') -> None:
     async def on_event(event: Event):
         if isinstance(event, AgentStateChangedObservation):
             if event.agent_state == AgentState.AWAITING_USER_INPUT:
-                message = input('Request user input >> ')
+                if fake_user_response_fn is None:
+                    message = input('Request user input >> ')
+                else:
+                    message = fake_user_response_fn(controller.get_state())
```

Comment:
I think add `fake_user_response_fn` into main function signature will cause test code and normal logic get coupled. But I also didn't find a good way to decouple it and mock the test. I prefer to merge it first and optimize it in the future. cc @li-boxuan  Do you have any good idea?

Reply:
This isn't for testing purpose - I believe it's for SWE-Bench evaluation, see [here](https://github.com/OpenDevin/OpenDevin/pull/1468/files#diff-cf5322e360ed33c3dc16023bfd443e6a7edab81c2f4a9e0dfd2f681a2fd59165R32-R37).

Looking at that giant PR, it looks like we could make this prompt as a built-in one?

> 'Please continue working on the task on whatever approach you think is suitable.\n'
        'If you think you have modified the code in a way that fixes the issue, please run the following command: <execute_bash> exit </execute_bash>.\n'
        'IMPORTANT: YOU SHOULD NEVER ASK FOR HUMAN HELP OR USE THE INTERNET TO SOLVE THIS TASK.\n'

The agent could have a `full-autonomous` mode so that it doesn't ask for user inputs, and instead, it uses the above prompt. @xingyaoww what do you think? I could help on that if you need a hand.

Reply:
@li-boxuan That's actually a great idea!! Fully support that -- I can help starts a PR to isolate that out - but the issue would be, do we want to implement this in the controller, or inside codeact agent like the count down?

Reply:
Ooh cool! The other agents have only "fully autonomous" mode currently, and need interactive mode, CodeAct the other way around. Maybe an option to set on the agent when run? 

Like `agent.mode=interactive` or `agent.mode=autonomous`. It is the agent who picks the prompt to use, isn't it?

Reply:
I tend to think the prompt itself will be in the agent itself but there should be an interface/api call from controller such that any agent can leverage this feature.

Reply:
Well i decided to keep it a separate PR after we got the SWE-Bench in -- supporting this now probably takes more time :(

Reply:
Sure, impelement feature first and refactor it when we have time, we also can help. :)